client.go

```diff
@@ -110,6 +114,15 @@ type Client struct {
 	// ResponseLogHook allows a user-supplied function to be called
 	// with the response from each HTTP request executed.
 	ResponseLogHook ResponseLogHook
+
+	// CheckRetry specifies a policy for handling retries. It is called
+	// following each request with the response and error values returned by
+	// the http.Client. If CheckRetry returns false, the Client stops retrying
+	// and returns the response to the caller. If CheckRetry returns an error,
+	// that error value is returned in lieu of the error from the request. If
+	// an error is returned, it is up to the CheckResponse callback to properly
+	// close any response body before returning.
+	CheckRetry func(resp *http.Response, err error) (bool, error)
```

Comment:
Fair point about the http package. I think we should be consistent though, and since this is an existing public package, we should probably keep the interfaces. Take a look at the RequestLogHook and ResponseLogHook, they are just function interfaces so function literals can still be used in the same way. I am in the habit of using an interface when multiple implementations are expected.